wig/wig.go

```diff
@@ -125,32 +126,40 @@ func Write(filename string, records map[string]Wig) {
 // WriteToFileHandle is a helper function for Write that writes the Wig data structure to an io.Writer
 func WriteToFileHandle(file io.Writer, rec Wig) {
 	var err error
-	if rec.StepType == "fixedStep" {
-		if rec.Span != -1 { //if there was a span in the header line
-			_, err = fmt.Fprintf(file, "%s chrom=%s start=%d step=%d span=%d\n", rec.StepType, rec.Chrom,
-				rec.Start, rec.Step, rec.Span)
+	var inDefaultRegion bool = true // bool to detect when we are in a region of defaultValues, where we don't need to write to file
+	for currIndex := range rec.Values {
+		if rec.Values[currIndex] != rec.DefaultValue {
+			if inDefaultRegion {
+				inDefaultRegion = false
+				_, err = fmt.Fprintf(file, "%s\n", makeWigHeaderString(rec, rec.Start+currIndex-1))
```

Comment:
makeWigHeaderString(rec, rec.Start+currIndex-1) calls makeWigHeaderString(record Wig, currIndex int) and prints a header line where start=currIndex+1. Would it be simpler to call makeWigHeaderString(rec, rec.Start+currIndex) and print start=currIndex?

Reply:
I thought this was more clear in that it makes it explicit we're converting between a 0-based and 1-based system. If I went with the solution you're describing, I would want to change the variable name away from 'currIndex', as to me 'index' implies 0 based, so start = currIndex looks a bit confusing.